### DISCUSSION TITLE
Is Redis a hard dependency?

### DISCUSSION THREAD

**Discussion by olimart** [AllYourBot/hostedgpt#257] [General]
Asking because SolidQueue is already installed and Render setup targets a paying plan ("starter").
So, maybe we can specifically target the free plan or just do it all with Postgres.
Also, Render does not allow to downgrade Redis plan at the moment.

**Comment by keithschacht** [comment]
@olimart Render has a free Redis option, right? I believe the `render.yml` file is configured to use that.

But regardless of that, it would be really nice to eliminate that dependency. When I first created this project the piece I was least familiar with was ActionCable. I followed a tutorial that used Redis for ActionCable but I've not investigated if postgres can maintain socket connections. Have you ever tried using postgres for actioncable?

Actually, one other notable thing is that the PR I just merged in last night used Redis. I was conscious that attempting to do a SQL read every 100 ms could be material on the DB so I thought I was being a little clever by writing this to the redis key-value store:
https://github.com/allyourbot/hostedgpt/commit/e89b46c539d73aeb1445b04ed13f306ea38da658

Basically, while the AI is replying and it's streaming to the browser, while it's in that streaming loop within the GetAIResponseJob it monitors for some redis value to change to know that it should abort early. This could easily be a configurable thing.

So the core question is really: configuring actionable to maintain socket connections with postgres and whether that's doable.

**Reply by keithschacht** [reply]
I just checked, it looks like the redis part of the config isn't specifying a plan. I'm not quite sure how that even works: https://github.com/allyourbot/hostedgpt/blob/main/render.yaml#L33

**Comment by olimart** [comment]
There is some limits with web sockets & pg at the moment for large payloads (I think).
There is also a gem that patches it (I think).
Rails 8 should resolve it though (I think).
In short, may be best to set the plan in Render config for now, or have kamal-ready deployment.

**Comment by keithschacht** [comment]
I'm updating the repo right now with the free plan.

**Comment by keithschacht** [comment]
@olimart I just finished eliminating Redis as a dependency! I love Rails' continuous focus on simplicity.

You might find this interesting: the queue was already using Postgres (SolidQueue FTW) and ActionCable had pre-existing support for Postgres as a backend, however it has an 8K limit. Based on discussions it looks like Rails 8 is going to solve this limitation but while the team debates what the right solution is, one of the proposed solutions has been spun out into a gem: https://github.com/reclaim-the-stack/actioncable-enhanced-postgresql-adapter

I just finished updating the two supported hosting options (Render and Fly) so they auto-deploy without Redis.

**Comment by olimart** [comment]
Fantastic @krschacht Thanks a lot
